### PR TITLE
Valo theme for datepicker

### PR DESCRIPTION
**Pull request by Toshegg**
Fixes https://github.com/vaadin/bakery-app-starter-polymer/issues/47

This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-themes/25)

### CODE REVIEW

File: demo/valo/common.html [hunk 1]
```diff
@@ -8,7 +8,8 @@
     {name: 'sizing-and-spacing', title: 'Sizing and spacing'},
     {name: 'buttons', title: 'Buttons', group: 'Components'},
     {name: 'checkboxes', title: 'Checkboxes'},
-    {name: 'text-fields', title: 'Text fields'}
+    {name: 'text-fields', title: 'Text fields'},
+    {name: 'date-pickers', title: 'Date pickers'}
```
Comment (jouni) [inline demo/valo/common.html]: Not set in stone, butI think alphabetical order would be better.

File: demo/valo/date-pickers.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+
+  <title>Valo Theme Date Picker</title>
+
+  <script src="../../../webcomponentsjs/webcomponents-loader.js"></script>
+  <link rel="import" href="common.html">
+
+  <link rel="import" href="../../valo/vaadin-date-picker.html">
+  <link rel="import" href="../../../vaadin-date-picker/vaadin-date-picker.html">
+  <link rel="import" href="../../../vaadin-date-picker-light.html">
```
Comment (jouni) [inline demo/valo/date-pickers.html]: vaadin-date-picker-light not used.

File: valo/vaadin-date-picker.html [hunk 3]
```diff
@@ -0,0 +1,172 @@
+<link rel="import" href="color.html">
+<link rel="import" href="sizing-and-spacing.html">
+<link rel="import" href="style.html">
+<link rel="import" href="typography.html">
+<link rel="import" href="vaadin-text-field.html">
+<!--<link rel="import" href="vaadin-button.html">-->
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Should be uncommented.

File: valo/vaadin-date-picker.html [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+<link href="https://fonts.googleapis.com/css?family=Cabin" rel="stylesheet">
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Unnecessary import.

File: valo/vaadin-date-picker.html [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+
+<dom-module id="date-picker-styles" theme-for="vaadin-date-picker">
+  <template>
+    <style include="vaadin-date-picker-default-theme">
+      :host {
+        font-family: 'Cabin', sans-serif;
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Wromg font. Also, I think it’s unnecessary to set the font here.

File: valo/vaadin-date-picker.html [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+<link rel="import" href="style.html">
+<link rel="import" href="typography.html">
+<link rel="import" href="vaadin-text-field.html">
+<!--<link rel="import" href="vaadin-button.html">-->
+<link href="https://fonts.googleapis.com/css?family=Cabin" rel="stylesheet">
+
+
+<dom-module id="date-picker-styles" theme-for="vaadin-date-picker">
+  <template>
+    <style include="vaadin-date-picker-default-theme">
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Do not include the default theme.

File: valo/vaadin-date-picker.html [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+      :host {
+        font-family: 'Cabin', sans-serif;
+        width: 250px;
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Should be computed. Use the same width as for text field.

File: valo/vaadin-date-picker.html [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+      [part="text-field"] {
+        max-width: 100%;
+      }
+    </style>
+  </template>
+</dom-module>
+
+<dom-module id="month-calendar-styles" theme-for="vaadin-month-calendar">
+  <template>
+    <style include="vaadin-month-calendar-default-theme">
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Do not include the default theme.

File: valo/vaadin-date-picker.html [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+      [part~="date"] {
+        padding: 10px 0;
+        font-size: 16px;
+        height: 42px;
+        width: 42px;
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Alot of magic numbers which should be replaced with computed values or variables.

File: valo/vaadin-date-picker.html [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+        font-size: 18px;
+        font-weight: 500;
+        color: #353f4c;
+      }
+    </style>
+  </template>
+</dom-module>
+<dom-module id="overlay-styles" theme-for="vaadin-date-picker-overlay">
+  <template>
+    <style include="vaadin-date-picker-overlay-default-theme">
```
Comment (jouni) [inline valo/vaadin-date-picker.html]: Do not include the default theme.

**Comment by manolo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 14 unresolved discussions.

---

*[valo/vaadin-date-picker.html, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTKkh8yWGdi1BtyDCg:-KoTKkh8yWGdi1BtyDCh:bahra3x) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L10)):*
> ```HTML
> 
> 
> <dom-module id="date-picker-styles" theme-for="vaadin-date-picker">
> ```

should be named as valo-date-picker-styles, otherwise we could have name collision 

---

*[valo/vaadin-date-picker.html, line 26 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTKrigSRR3odQCRkoW:-KoTKrigSRR3odQCRkoX:btqaop0) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L26)):*
> ```HTML
> </dom-module>
> 
> <dom-module id="month-calendar-styles" theme-for="vaadin-month-calendar">
> ```

rename this as well, prepend `valo-`

---

*[valo/vaadin-date-picker.html, line 106 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTKzBKGallaOzTkjji:-KoTKzBKGallaOzTkjjj:b-dmvrm7) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L106)):*
> ```HTML
>   </template>
> </dom-module>
> <dom-module id="overlay-styles" theme-for="vaadin-date-picker-overlay">
> ```

rename this as well, prepend `valo-vaadin-date-picker`

---

*[valo/vaadin-date-picker.html, line 151 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTL9MqsFAuL7DdGwoM:-KoTL9MqsFAuL7DdGwoN:bz9sa7z) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L151)):*
> ```HTML
>   </template>
> </dom-module>
> <dom-module id="button-styles" theme-for="vaadin-button">
> ```

needs renaming as well

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25)*

**Comment by manolo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 14 unresolved discussions.

---

*[valo/vaadin-date-picker.html, line 151 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTL9MqsFAuL7DdGwoM:-KoTMsY9uy5BRveYaU3B:b-yc2tzu) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L151)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

needs renaming as well
</blockquote></details>

Actually we don't need this module at all, it would modify all buttons in the app. Date-picker should use valo styled buttons directly. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25)*

**Comment by Toshegg** [comment]
Review status: 0 of 4 files reviewed at latest revision, 14 unresolved discussions.

---

*[demo/valo/common.html, line 12 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL2-r1-12:-Kog6hFo7rmt68_eutyg:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/demo/valo/common.html#L12)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Not set in stone, butI think alphabetical order would be better.
</blockquote></details>

Done.

---

*[demo/valo/date-pickers.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL3-r1-15:-Kog6lPCpz30v2NN_I7v:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/demo/valo/date-pickers.html#L15)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

vaadin-date-picker-light not used. 
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 6 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-6:-Kog6yj-i20ke0lWtrhI:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L6)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Should be uncommented.
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 7 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-7:-Kog70WAIgXqrPZtBTUq:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L7)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Unnecessary import. 
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTKkh8yWGdi1BtyDCg:-Kog7AC_e4hTcpIdktIr:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L10)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

should be named as valo-date-picker-styles, otherwise we could have name collision 
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 12 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-12:-Kog79mywNNq-lCpy0Lt:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L12)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Do not include the default theme.
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 14 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-14:-KolOp-SaPttejCE--ns:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L14)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Wromg font. Also, I think it’s unnecessary to set the font here.
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-15:-KolyKEWBCsKVcs37kd1:bs7wumb) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L15)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Should be computed. Use the same width as for text field.
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 26 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTKrigSRR3odQCRkoW:-Kog7E1JaTffYYNvOE9V:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L26)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

rename this as well, prepend `valo-`
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 28 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-28:-Kog7E_Zip_z4098wJjL:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L28)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Do not include the default theme.
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 33 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-33:-Kom0HPGYHI2yJQ_a99v:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L33)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Alot of magic numbers which should be replaced with computed values or variables.
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 106 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTKzBKGallaOzTkjji:-Kog7XX6aIwaK5tRJjwP:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L106)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

rename this as well, prepend `valo-vaadin-date-picker`
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 108 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoSui5yAbAuQ7OkAzL5-r1-108:-Kog7Xy-jEQ3iLHgYpAU:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L108)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Do not include the default theme. 
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 151 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KoTL9MqsFAuL7DdGwoM:-KolxBG40wC4Wv9xyRBQ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/6436c7df0b494e7a6c8ece679c1415ab4cac556e/valo/vaadin-date-picker.html#L151)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Actually we don't need this module at all, it would modify all buttons in the app. Date-picker should use valo styled buttons directly. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25)*

**Comment by manolo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

*[valo/vaadin-date-picker.html, line 19 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KomZNslG4EO4_MklA9E:-KomZNsmdMfJGBY4nJIl:be3axy7) ([raw file](https://github.com/vaadin/vaadin-themes/blob/adc8a5def8e5c2ca49d12bf22e2b7eb37878c08c/valo/vaadin-date-picker.html#L19)):*
> ```HTML
>       [part~="overlay"] {
>         border-radius: 6px;
>         box-shadow: 0 1px 4px 0 rgba(31, 70, 120, 0.3);
> ```

could you pickup some valo color instead ?

---

*[valo/vaadin-date-picker.html, line 37 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KomYwdJumwWrMtLcKKc:-KomYwdKU6oAeSZByTW5:b-xurn8d) ([raw file](https://github.com/vaadin/vaadin-themes/blob/adc8a5def8e5c2ca49d12bf22e2b7eb37878c08c/valo/vaadin-date-picker.html#L37)):*
> ```HTML
>         -webkit-user-select: none;
>         user-select: none;
>         --valo-month-calendar-date-font-size: 16px;
> ```

Probably this font sizes should use valo font sizes from https://cdn.vaadin.com/vaadin-themes/0.2.0/demo/valo/index.html#font-size 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25)*

**Comment by Toshegg** [comment]
Review status: 0 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

*[valo/vaadin-date-picker.html, line 19 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KomZNslG4EO4_MklA9E:-KonDehz4aRerUVywEoQ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/adc8a5def8e5c2ca49d12bf22e2b7eb37878c08c/valo/vaadin-date-picker.html#L19)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

could you pickup some valo color instead ?
</blockquote></details>

Done.

---

*[valo/vaadin-date-picker.html, line 37 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-KomYwdJumwWrMtLcKKc:-KonDfQ-TlKI18muvcNv:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themes/blob/adc8a5def8e5c2ca49d12bf22e2b7eb37878c08c/valo/vaadin-date-picker.html#L37)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Probably this font sizes should use valo font sizes from https://cdn.vaadin.com/vaadin-themes/0.2.0/demo/valo/index.html#font-size 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25)*

**Comment by manolo** [comment]
I think the gap between months should be reduced at least compared with the https://app.zeplin.io/project/5937d86d3601d452f1b8e66a/screen/593a69fddec2a8e38b128100 

---

Review status: 0 of 4 files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-:-KoqeN3T65I73npr4hs2:b-sz8a49)*

**Comment by manolo** [comment]
:lgtm:

---

Reviewed 1 of 4 files at r1, 2 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25#-:-KoqoD2M3UsdTmqJ7AX0:bnfp4nl)*

**Comment by limonte** [comment]
- 'clear field' icon should have `cursor: pointer`
- placeholder should not be `YYYY-MM-DD`, it's not the correct format, keep aligned with material demo page and use the `Pick a date` placeholder
- `id="weeks"` is not needed in the example
- no need to use `show-week-numbers` like `show-week-numbers=""`, see https://cdn.vaadin.com/vaadin-core-elements/master/vaadin-date-picker/demo/index.html#date-picker-with-week-numbers 
- no need in `opened=""`, keep aligned with material demo page

**Comment by limonte** [comment]
- `today` and `cancel` buttons should have `cursor: pointer`

**Comment by manolo** [comment]
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/25)*

**Pull request merged**